generic_config_updater/generic_updater.py

```diff
@@ -86,64 +86,123 @@ def apply(self, patch):
 
 class ConfigReplacer:
     def __init__(self, patch_applier=None, config_wrapper=None, patch_wrapper=None):
+        self.logger = genericUpdaterLogging.get_logger(title="Config Replacer")
         self.patch_applier = patch_applier if patch_applier is not None else PatchApplier()
         self.config_wrapper = config_wrapper if config_wrapper is not None else ConfigWrapper()
         self.patch_wrapper = patch_wrapper if patch_wrapper is not None else PatchWrapper()
 
     def replace(self, target_config):
+        print_to_console=True
+        self.logger.log_notice("Config replacement starting.", print_to_console)
```

Comment:
>print_to_console [](http://example.com/codeflow?start=63&length=16)

It seems that you often want to print_to_console consistently per class. I am wondering if you can add it as a `get_logger` optional prameter, and the default value is None. If provide `True` or `False`, it will override each `log_*` option.

Reply:
Good idea!

I simplified the flag to `print_all_to_console` which is passed to get_logger.

If `print_all_to_console` is True: all lines are printed regardless of the input flag to `log_*` methods.
If `print_all_to_console` is False: all lines are printed according to the input flag to `log_*` methods.